### PR TITLE
Callback

### PR DESCRIPTION
**Pull request by DRigby26**

### CODE REVIEW

File: src/services/dragonchain-client/DragonchainClient.ts [hunk 1]
```diff
@@ -508,8 +509,8 @@ export class DragonchainClient {
   /**
    * @hidden
    */
-  private async makeRequest (path: string, method: SupportedHTTP, body: string = '', jsonParse: boolean = true) {
-    const fetchData = this.getFetchOptions(method, path, body)
+  private async makeRequest (path: string, method: SupportedHTTP, callbackURL?: string, body: string = '', jsonParse: boolean = true) {
```
Comment (cheeseandcereal) [inline src/services/dragonchain-client/DragonchainClient.ts]: It's anti-pattern to mix optional and default params together

File: src/services/dragonchain-client/DragonchainClient.ts [hunk 2]
```diff
@@ -472,14 +472,15 @@ export class DragonchainClient {
   /**
    * @hidden
    */
-  private getFetchOptions (method: SupportedHTTP, path: string, body: string, contentType: string = 'application/json'): FetchOptions {
+  private getFetchOptions (method: SupportedHTTP, path: string, body: string, callbackURL?: string, contentType: string = 'application/json'): FetchOptions {
```
Comment (cheeseandcereal) [inline src/services/dragonchain-client/DragonchainClient.ts]: It's anti-pattern to mix optional and default params together

File: src/services/dragonchain-client/DragonchainClient.ts [hunk 3]
```diff
@@ -508,8 +509,8 @@ export class DragonchainClient {
   /**
    * @hidden
    */
-  private async makeRequest (path: string, method: SupportedHTTP, body: string = '', jsonParse: boolean = true) {
-    const fetchData = this.getFetchOptions(method, path, body)
+  private async makeRequest (path: string, method: SupportedHTTP, callbackURL?: string, body: string = '', jsonParse: boolean = true) {
+    const fetchData = this.getFetchOptions(method, path, body, callbackURL)
```
Comment (deanshelton913) [inline src/services/dragonchain-client/DragonchainClient.ts]: Total nit-pick:
Can we even out the `getFetchOptions` arguments to match the order of `makeRequest`? This looks confusing, and took me a sec.
Suggest we just re-arrange
```suggestion
    const fetchData = this.getFetchOptions(path, method, callbackUrl, body)

**Comment by deanshelton913** [comment]
plz make sure the docs show the new argument

File: src/services/dragonchain-client/DragonchainClient.ts (hunk 2 above)
Comment (deanshelton913) [reply src/services/dragonchain-client/DragonchainClient.ts]: Dont disagree. Can you elaborate on your note here?
AFAIK: optional params are supposed to go at the end, right?
This signature has two optional params, one defaulting to `undefined` and the other defaulting to a string value.

**Pull request merged**